AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/ConfigurationTests/AWSCognitoAuthPluginConfigTests.swift

```diff
@@ -188,4 +188,61 @@ class AWSCognitoAuthPluginConfigTests: XCTestCase {
         }
     }
 
+    /// Test Auth configuration order of execution of apis
+    ///
+    /// - Given: Given valid config for user pool and identity pool
+    /// - When:
+    ///    - I call different auth apis after configuration
+    /// - Then:
+    ///    - I should get the result back based on the order in which the calls where made.
+    ///
+    func testAPIExecutionOrder() throws {
+        let plugin = AWSCognitoAuthPlugin()
+        try Amplify.add(plugin: plugin)
+
+        let categoryConfig = AuthCategoryConfiguration(plugins: [
+            "awsCognitoAuthPlugin": [
+                "CredentialsProvider": ["CognitoIdentity": ["Default":
+                    ["PoolId": "xx",
+                     "Region": "us-east-1"]
+                    ]],
+                "CognitoUserPool": ["Default": [
+                    "PoolId": "xx",
+                    "Region": "us-east-1",
+                    "AppClientId": "xx",
+                    "AppClientSecret": "xx"]]
+            ]
+        ])
+        let amplifyConfig = AmplifyConfiguration(auth: categoryConfig)
+        do {
+            try Amplify.configure(amplifyConfig)
+        } catch {
+            XCTFail("Should not throw error. \(error)")
+        }
+
+        let resultExpectation1 = expectation(description: "Should receive a result")
+        _ = plugin.signUp(username: "mockUsername", password: "", options: nil) { _ in
+            resultExpectation1.fulfill()
+        }
+
+        var expectationList = [resultExpectation1]
+
+        for _ in 1 ... 50 {
+            let resultExpectation2 = expectation(description: "Should receive a result")
```

Comment:
naming nit: this is actually resultExpectation(2...51) right? :)

Reply:
Fixed